pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -402,6 +406,10 @@ func (e *etcd) Deploy(ctx context.Context) error {
 		networkingv1.NetworkPolicyPort{Port: utils.IntStrPtrFromInt(int(PortEtcdClient)), Protocol: utils.ProtocolPtr(corev1.ProtocolTCP)},
 		networkingv1.NetworkPolicyPort{Port: utils.IntStrPtrFromInt(int(PortBackupRestore)), Protocol: utils.ProtocolPtr(corev1.ProtocolTCP)},
 	))
+	if e.values.TopologyAwareRoutingEnabled {
+		metav1.SetMetaDataAnnotation(&clientService.ObjectMeta, corev1.AnnotationTopologyAwareHints, "auto")
+		metav1.SetMetaDataLabel(&clientService.ObjectMeta, resourcesv1alpha1.EndpointSliceHintsConsider, "true")
```

Comment:
What about a function `MakeRoutingTopologyAware` like `MakeUnique` which takes the service and sets the annotations and labels? That's easier than duplicating this for each and every service?

Reply:
Should I move only 
```golang
			metav1.SetMetaDataAnnotation(&service.ObjectMeta, corev1.AnnotationTopologyAwareHints, "auto")
			metav1.SetMetaDataLabel(&service.ObjectMeta, resourcesv1alpha1.EndpointSliceHintsConsider, "true")
```

to an utility func or should I move

```golang
		if topologyAwareRoutingEnabled {
			metav1.SetMetaDataAnnotation(&service.ObjectMeta, corev1.AnnotationTopologyAwareHints, "auto")
			metav1.SetMetaDataLabel(&service.ObjectMeta, resourcesv1alpha1.EndpointSliceHintsConsider, "true")
		} else {
			delete(service.Annotations, corev1.AnnotationTopologyAwareHints)
			delete(service.Labels, resourcesv1alpha1.EndpointSliceHintsConsider)
		}
```

?

Reply:
The latter seems reasonable (this will be needed everywhere as well since it's configurable).

Reply:
I now moved the above 2nd handling to a dedicated utility function.